launchable/utils/git_log_parser.py

```diff
@@ -0,0 +1,50 @@
+from collections import namedtuple
+from typing import Dict, List, TextIO, Any
+from datetime import datetime
+import dateutil.parser
+import json
+
+ChangedFile = namedtuple('ChangedFile', ['path', 'added', 'deleted'])
+
+GitCommit = namedtuple('GitCommit', [
+    'commit_hash', 'parents', 'author_email', 'author_time', 'committer_email',
+    'committer_time', 'changed_files'
+])
+
+
+def parse_git_log(fp: TextIO) -> List[GitCommit]:
+    """Parses the output of a git log command.
+
+    This parses the output of `git log --pretty='format:{"commit": "%H",
+    "parents": "%P", "authorEmail": "%ae", "authorTime": "%aI",
+    "committerEmail": "%ce", "committerTime": "%cI"}' --numstat`
+    """
+    ret = []
+    meta = {}  # type: Dict[str, Any]
```

Comment:
[imo]
it can be written like
```python
meta: Dict[str, Any] = {}
```

Reply:
Note that this needs to be compatible with Python 3.5